homeassistant/components/media_player/__init__.py

```diff
@@ -352,12 +358,20 @@ def select_source_service(service):
                            schema=MEDIA_PLAYER_SELECT_SOURCE_SCHEMA)
 
     def play_media_service(service):
-        """Play specified media_id on the media player."""
+        """
+        Play specified media_id on the media player.
+
+        If ATTR_MEDIA_ENQUEUE is true, add media_id to the queue instead.
+        """
         media_type = service.data.get(ATTR_MEDIA_CONTENT_TYPE)
         media_id = service.data.get(ATTR_MEDIA_CONTENT_ID)
+        enqueue = service.data.get(ATTR_MEDIA_ENQUEUE)
 
         for player in component.extract_from_service(service):
-            player.play_media(media_type, media_id)
+            if player.support_enqueue:
+                player.play_media(media_type, media_id, enqueue)
```

Comment:
This should become

```python
kwargs = {
    ATTR_MEDIA_ENQUEUE: enqueue,
}
player.play_media(media_type, media_id, **kwargs)
```

All play_media methods on devices need to be updated to accept `**kwargs`. 

In Sonos.play_media you can then do:

```python
enqueue = kwargs.get(ATTR_MEDIA_ENQUEUE)
```